api/src/main/proto/stellarstation/api/v1/monitoring/monitoring.proto

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2018 Infostellar, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+import "stellarstation/api/v1/antenna/antenna.proto";
+import "stellarstation/api/v1/common/common.proto";
+import "stellarstation/api/v1/radio/radio.proto";
+
+package stellarstation.api.v1.monitoring;
+
+option go_package = "github.com/infostellarinc/go-stellarstation/api/v1/monitoring";
+
+option java_multiple_files = true;
+option java_outer_classname = "MonitoringProto";
+option java_package = "com.stellarstation.api.v1.monitoring";
+
+// Messages for reporting information about a ground station for monitoring and debugging.
+
+// The current configuration of a transmitter. A transmitter is taking a bitstream and applies
+// signal processing to create the final waveform sent to a satellite.
+message TransmitterConfiguration {
+  // The modulation type of the transmitter.
+  radio.Modulation modulation = 1;
+
+  // Whether the carrier is enabled.
+  bool is_carrier_enabled = 2;
+
+  // The bitrate of the transmitter.
+  uint64 bitrate = 3;
+
+  // The line coding of the transmitter.
+  radio.LineCoding line_coding = 4;
+
+  // The bandwidth of the filter.
+  uint64 filter_bandwidth_hz = 5;
+
+  // The sampling frequency.
+  uint64 sampling_frequency_hz = 6;
+}
+
+// The current configuration of a receiver. A receiver is taking a waveform from a satellite and
+// applying signal processing to produce a bitstream.
+message ReceiverConfiguration {
+  // The modulation type of the receiver.
+  radio.Modulation modulation = 1;
+
+  // The bandwidth, in Hz.
+  uint64 bandwidth_hz = 2;
+
+  // The sampling frequency.
+  uint64 sampling_frequency_hz = 3;
+
+  // The bitrate of the receiver.
+  uint64 bitrate = 4;
+
+  // The line coding of the receiver.
+  radio.LineCoding line_coding = 5;
+
+  // The frame synchronization word being used. Unset if frame synchronization is disabled.
+  common.Bits synchronization_word = 6;
+
+  // Whether convolutional code descrambling is enabled.
+  bool is_convolutional_code_descrambling = 7;
```

Comment:
Are all of these booleans independent?

Reply:
I think inverted / decoding are linked, added a comment.